api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java

```diff
@@ -1141,7 +1141,19 @@ public ConceptSource getConceptSourceByName(String conceptSourceName) throws DAO
 		criteria.add(Restrictions.eq("source.name", conceptSourceName));
 		return (ConceptSource) criteria.uniqueResult();
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.ConceptDAO#getConceptSourceByUniqueId(java.lang.String)
+	 */
+	public ConceptSource getConceptSourceByUniqueId(String uniqueId) {
+		if (StringUtils.isBlank(uniqueId)) {
+			return null;
```

Comment:
I am totally for failing fast. But I think we might need to split it. Passing null leads to an exception and passing an empty string you get null because an empty string is more like passing a string that doesn't exist. Right?